### PR TITLE
MEV plugin with Merge integration

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.Consensus/ManualGasLimitCalculator.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using Nethermind.Core;
+
+namespace Nethermind.Consensus
+{
+    public class ManualGasLimitCalculator : IGasLimitCalculator
+    {
+        public long GasLimit { get; set; }
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Consensus/ManualGasLimitCalculator.cs:24]: Hmmm what about 1559 here? Let's discuss

File: src/Nethermind/Nethermind.Core/Collections/LinkedHashSet.cs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+// 
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nethermind.Core.Collections
+{
+    public class LinkedHashSet<T> : ISet<T> where T : notnull
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Core/Collections/LinkedHashSet.cs:25]: Do we have tests for this class?

File: src/Nethermind/Nethermind.Core/Collections/SortedRealList.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Core/Collections/SortedRealList.cs:19]: Do we have tests for this class?

File: src/Nethermind/Nethermind.Consensus/ManualGasLimitCalculator.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Consensus/ManualGasLimitCalculator.cs:24]: Only used in tests for now.

File: src/Nethermind/Nethermind.Core/Collections/SortedRealList.cs (hunk 3 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Core/Collections/SortedRealList.cs:19]: No, this class adds no functionality

**Review by marcindsobczak — APPROVED** [review]

**Review by MarekM25 — APPROVED** [review]

**Pull request merged**